validator/delphi_validator/validate.py

```diff
@@ -83,11 +83,9 @@ def __init__(self, params):
         self.suppressed_errors = {(item,) if not isinstance(item, tuple) and not isinstance(
             item, list) else tuple(item) for item in params.get('suppressed_errors', [])}
 
-        self.active_report = ValidationReport(self.suppressed_errors)
```

Comment:
whats the rational for moving this from being a class attribute? A Validator object maintaining it's own ValidatorReport doesn't seem unreasonable

Reply:
The original point of having it as its own attribute was that then you didn't have to pass it as an argument to all its function calls for individual checks.  Now that the checks are in separate objects/files, that advantage is no longer there.  The disadvantage of having it be an attribute is that it unnecessarily introduces an internally persistent state (need to be sure nothing changes between initialization and the actual validation call).  Essentially we are considering two possibilities now:

```
class X:
  def __init__(self):
    self.y = None
  def add(self):
    return self.y + 3
  def run(self):
    self.y = Y()
    return self.do_something()
```
and
```
class X:
  def add(self, y):
    return y + 3
  def run(self):
    y = Y()
    return self.do_something(y)
```

The latter seems much better to me.

Reply:
sgtm